scripts/Stretch.py

```diff
@@ -0,0 +1,339 @@
+        current_y = 0.
+        current_z = 0.
+        current_e = 0.
+        current_f = 0.
+
+        self.output_x = 0.
+        self.output_y = 0.
+        self.output_z = 0.
+        self.output_e = 0.
+        self.output_f = 0.
+
+        layer_z = 0.
+        for layer in data:
+            lines = layer.rstrip("\n").split("\n")
+            for line in lines:
+                comment = ""
+                if line.find(';') >= 0:
+                    comment = line[line.find(';'):]
+                if self.getValue(line,'G') == 0:
+                    current_x = self.getValue(line,'X',current_x)
+                    current_y = self.getValue(line,'Y',current_y)
+                    current_z = self.getValue(line,'Z',current_z)
+                    current_e = self.getValue(line,'E',current_e)
+                    current_f = self.getValue(line,'F',current_f)
+                    onestep = GCodeStep(0,current_x,current_y,current_z,current_e,current_f,comment)
+                elif self.getValue(line,'G') == 1:
+                    current_x = self.getValue(line,'X',current_x)
+                    current_y = self.getValue(line,'Y',current_y)
+                    current_z = self.getValue(line,'Z',current_z)
+                    current_e = self.getValue(line,'E',current_e)
+                    current_f = self.getValue(line,'F',current_f)
+                    onestep = GCodeStep(1,current_x,current_y,current_z,current_e,current_f,comment)
+                elif self.getValue(line,'G') == 92:
+                    current_x = self.getValue(line,'X',current_x)
+                    current_y = self.getValue(line,'Y',current_y)
+                    current_z = self.getValue(line,'Z',current_z)
+                    current_e = self.getValue(line,'E',current_e)
+                    current_f = self.getValue(line,'F',current_f)
+                    onestep = GCodeStep(-1,current_x,current_y,current_z,current_e,current_f,comment)
+                else:
+                    onestep = GCodeStep(-1,current_x,current_y,current_z,current_e,current_f,line);
+                if current_z != layer_z:
+                    print("Layer Z " + "{:.3f}".format(layer_z) + " " + str(len(LayerSteps)) + " steps")
+                    if len(LayerSteps):
+                        retdata.append(self.ProcessLayer(LayerSteps))
+                    LayerSteps = []
+                    layer_z = current_z
+                LayerSteps.append(onestep)
+        if len(LayerSteps):
+            retdata.append(self.ProcessLayer(LayerSteps))
+        return retdata
+    def ProcessLayer(self,LayerSteps):
+        layergcode = ""
+        self.vd1 = np.empty((0,2)) #Start points of segments of already deposited material for this layer
+        self.vd2 = np.empty((0,2)) #End points of segments of already deposited material for this layer
+        curE = LayerSteps[0].e
+        vPos = np.empty((0,2))
+        iflush = 0
+        for i in range(len(LayerSteps)):
```

Comment:
You can use enumerate for this. Example; for i, step in enumerate(layer_steps):
This gives you both acces to the current step and the index.